### PR TITLE
Ballistic Weapon/Shotgun Reload Sounds

### PR DESCRIPTION
**Pull request by Steelpoint**
This PR adds multiple new sounds to many ballistic weapons. Sounds trigger when you: Take a bullet out of a mag/revolver, insert a bullet into a mag/revolver, place a new magazine into a weapon, insert a new shell into a shotgun and when you open, close and remove the box of a SAW Machinegun.

Here is a link to a video showcasing the new sounds: https://www.dropbox.com/s/xwmafq4a4rtwr9v/Desktop%2023.MAY.2017%20-%2022.24.09.04.mp4?dl=0

:cl: Steelpoint
add: Many Ballistic weapons now have new sounds related to reloading or placing bullets into magazines.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/ballistic/automatic.dm [hunk 1]
```diff
@@ -44,6 +44,8 @@
 					oldmag.update_icon()
 				else
 					to_chat(user, "<span class='notice'>You insert the magazine into \the [src].</span>")
+
+				playsound(user, 'sound/weapons/autoguninsert.ogg', 60, 1)
```
Comment (Steelpoint) [inline code/modules/projectiles/guns/ballistic/automatic.dm]: I have a issue here where the sound only plays when someone performs a 'tac reload' and not placing the magazine when the gun has no mag present.

**Comment by Bawhoppen** [comment]
someone will find a way to downvote this

**Comment by WJohn** [comment]
Been wanting this for a long time. 👍

**Comment by Steelpoint** [comment]
My only issue is that the automatic ballistic reload sound only currently plays for when you do a tac reload, not when you reload when the gun has no mag in it.

**Comment by WJohn** [comment]
You could add that, as well as a sound for loading a bullet into a magazine, and for using a speed mag on a revolver. Putting a single bullet in the revolver should use the same one as a bullet in a magazine, of course.

**Comment by Steelpoint** [comment]
I would need a sound for that, I could use the current 'clicking' sound tho.

I just need to figure out why the reload sound will only play for tac reloading and not normal reloading.

**Comment by mattroks101** [comment]
Would it be too much trouble just to make an unload/reload system like what old Hippie used to have? So guns could have different unloading and reloading sounds?

**Comment by Steelpoint** [comment]
I can't even get the gun sounds to trigger when loading a empty magazine let alone a complex system allowing unique reload sounds for different guns.

Also that would require a lot of sound assets.

**Comment by mattroks101** [comment]
Hippie just had a really simple system of guns have an unload and reload sound var, and then just playing that sound whenever you unload or reload. I ported it to my fork of TG a while ago. But you're right about the sound asset part.

**Comment by Steelpoint** [comment]
Still have not figured out why the sounds won't play correctly when reloading a empty gun with no mag.

**Comment by Steelpoint** [comment]
Going to try and add some extra sounds for other gun related actions.

**Comment by Steelpoint** [comment]
I've updated this with more gun related sounds.

I've also made a quick video showcasing these changes, biggest issue remains how the game won't play a sound when reloading a empty mag, but everything else so far is good. 

Here's the video, which is also in the OP, shitty quality aside: https://www.dropbox.com/s/xwmafq4a4rtwr9v/Desktop%2023.MAY.2017%20-%2022.24.09.04.mp4?dl=0

**Comment by Tacolizard** [comment]
@Steelpoint extra good boy points if you add more gun firing sounds based on caliber

**Comment by Steelpoint** [comment]
That would require not only a more complicated code system to differentiate between different guns and what calibre they fire, but it would also require a lot more sound assets for each firing sound. 

It took me long enough to find acceptable sounds that I could freely use as it is. I think the only other way we can get more sound files is if someone either makes them for us or rips them from another game and we're allowed to use them.

Not to mention I still cannot get the gun to play a reload sound when you load a mag into a gun with no mag in it.

**Pull request merged**